### PR TITLE
[DataGridPro] Fix double top border in header filters

### PR DESCRIPTION
**Pull request by MBilalShafi**
Spotted while working on #14373 

| Before | After |
| --- | --- |
| image | image |

Before Preview: https://deploy-preview-14366--material-ui-x.netlify.app/x/react-data-grid/filtering/header-filters/
After Preview: https://deploy-preview-14375--material-ui-x.netlify.app/x/react-data-grid/filtering/header-filters/

### Fixes

1. Double top border in header filters
2. Messed up scrollbar fillers' borders

### CODE REVIEW

**Review by KenanYusuf — APPROVED** [review]
Nice spot, looks good to me.

File: packages/x-data-grid/src/components/containers/GridRootStyles.ts
Comment (romgrk) [inline packages/x-data-grid/src/components/containers/GridRootStyles.ts:1]: Minor issue, if you look close enough the border doesn't extend all the way over the vertical scrollbar. If you address this, also make sure it looks fine with top pinned rows.

image

Comment (romgrk) [reply packages/x-data-grid/src/components/containers/GridRootStyles.ts:1]: With the new changes it messes the border in bottom pinned rows. I don't think we should be removing the top-border where there is one, it seems like we should be adding a bottom-border where one is needed.

image

Comment (MBilalShafi) [reply packages/x-data-grid/src/components/containers/GridRootStyles.ts:1]: > With the new changes it messes the border in bottom pinned rows. I don't think we should be removing the top-border where there is one, it seems like we should be adding a bottom-border where one is needed.

Yes, I noted it too. On it. Thanks for identifying.

Comment (MBilalShafi) [reply packages/x-data-grid/src/components/containers/GridRootStyles.ts:1]: Did some updates, and it looks good to me now, tested in https://codesandbox.io/p/sandbox/mui-mui-x-x-data-grid-42yp27

**Review by romgrk — APPROVED** [review]

**Pull request merged**